bin/dist_get

```diff
@@ -14,63 +14,65 @@ check_writeable() {
 }
 
 download() {
-	local url="$1"
-	local output="$2"
+	dl_url="$1"
+	dl_output="$2"
 
-	if [ -z "$url" ] || [ -z "$output" ]; then
-		die "download takes exactly two arguments. was given '$@'"
-	fi
+	test "$#" -eq "2" || die "download requires exactly two arguments, was given $@"
 
 	if ! check_writeable "$output"; then
 		die "download error: cannot write to $output"
 	fi
 
 	if have_binary wget; then
-		printf 'Using wget to download "%s" to "%s"\n' "$url" "$output"
+		printf '==> Using wget to download "%s" to "%s"\n' "$url" "$output"
 		wget "$url" -O "$output"
```

Comment:
If you use `wget "$url" -O "$output" || return` here and something like that for curl and fetch below, then you don't need `if [ "$?" -ne 0 ]; then return $?; fi` after the if, and it will do the right thing, which is return the wget error code.